### PR TITLE
Contig support

### PR DESCRIPTION
**Pull request by cbun**
Added support in WASP engine for a contig file type.  To be used with recipes/plugins that can use  `self.data.contigfiles`.  

Added a `contigs_compare`recipe that runs quast on all contig files: 
```arast run --contigs contigs1.fa --contigs2.fa --recipe contigs_compare```

Newest version of `pymongo` uses a different connection constructor.  Updated to support that.

### CODE REVIEW

File: lib/assembly/client.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def get_config(self):
 
     def req(self, url, req_type='get', data=None, ret=None):
         """Authenticated request. Parses CherryPy message and raises HTTPError"""
-        # print "req_{}: {}".format(req_type, url)
+        print "req_{}: {}".format(req_type, url)
```
Comment (levinas) [inline lib/assembly/client.py]: Is this for debugging?

File: lib/assembly/consume.py [hunk 2]
```diff
@@ -487,6 +491,7 @@ def download_url(self, url, outdir, token=None):
         return self.extract_file(downloaded)
 
     def fetch_job(self):
+        print 'fetch'
```
Comment (levinas) [inline lib/assembly/consume.py]: There's a line with `logger.info('Fetching job...')` a few lines down.

File: lib/assembly/plugins.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ class BasePlugin(object):
     """
 
     def base_call(self, settings, job_data, manager, strict=False):
+        print 'base call'
```
Comment (levinas) [inline lib/assembly/plugins.py]: use logger.debug?

File: lib/assembly/plugins.py [hunk 4]
```diff
@@ -91,7 +92,7 @@ def arast_popen(self, cmd_args, overrides=True, **kwargs):
             cmd_string = human_readable_command(cmd_args)
         else:
             cmd_string = cmd_args
-
+        print 'PLUGIN:', cmd_string
```
Comment (levinas) [inline lib/assembly/plugins.py]: logger

File: lib/assembly/plugins/quast.py [hunk 5]
```diff
@@ -41,9 +41,10 @@ def run(self):
 
         #### Add Contig files ####
         cmd_args += contigfiles
-        cmd_args += ['-l', '"{}"'.format(', '.join([cset.name for cset in contigsets]))]
-
-
+        try:
+            cmd_args += ['-l', '"{}"'.format(', '.join([cset.name for cset in contigsets]))]
+        except (KeyError, TypeError): # No name
+            pass
```
Comment (levinas) [inline lib/assembly/plugins/quast.py]: What do these exceptions mean?

File: lib/assembly/shock.py [hunk 6]
```diff
@@ -238,6 +238,7 @@ def _post_file(self, filename, filetype='', auth=False):
             raise Error("python-requests error: {}. Try with --curl flag.".format(e))
 
         attr_fd.close()
+        print r.text
```
Comment (levinas) [inline lib/assembly/shock.py]: logger

File: lib/assembly/plugins/quast.py (hunk 5 above)
Comment (cbun) [inline lib/assembly/plugins/quast.py]: With the new contig file type, uploaded contigs don't get named, so quast was breaking trying to get label names from ContigSets.

**Comment by cbun** [comment]
@levinas I addressed your comments

**Pull request merged**